cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaNodeSetController.java

```diff
@@ -0,0 +1,478 @@
+        if (!new StatusDiff(nodeSet.getStatus(), desiredStatus).isEmpty())  {
+            try {
+                LOGGER.debugCr(reconciliation, "Updating status of KafkaNodeSet {} in namespace {}", reconciliation.name(), reconciliation.namespace());
+                KafkaNodeSet updatedNodeSet = new KafkaNodeSetBuilder(kafkaNodeSetLister.namespace(reconciliation.namespace()).get(reconciliation.name()))
+                        .withStatus(desiredStatus)
+                        .build();
+
+                kafkaNodeSetOperator.client().inNamespace(reconciliation.namespace()).patchStatus(updatedNodeSet);
+            } catch (KubernetesClientException e)   {
+                LOGGER.errorCr(reconciliation, "Failed to update status of KafkaNodeSet {} in namespace {}", reconciliation.name(), reconciliation.namespace(), e);
+            }
+        }
+    }
+
+    /**
+     * Creates missing pod defined in the KafkaNodeSet
+     *
+     * @param reconciliation    Reconciliation in which this is executed
+     * @param pod               Pod which should be checked and created if needed
+     * @param owner             The OwnerReference which should be set to the pod
+     * @param podCounter        Pod Counter used to count pods for the status
+     */
+    private void maybeCreatePod(Reconciliation reconciliation, Pod pod, OwnerReference owner, PodCounter podCounter)    {
+        Pod currentPod = podLister.namespace(reconciliation.namespace()).get(pod.getMetadata().getName());
+
+        if (currentPod == null) {
+            // Pod does not exist => we create it
+            LOGGER.debugCr(reconciliation, "Creating pod {} in namespace {}", pod.getMetadata().getName(), reconciliation.namespace());
+            pod.getMetadata().setOwnerReferences(List.of(owner));
+            podOperator.client().inNamespace(reconciliation.namespace()).create(pod);
+        } else {
+            LOGGER.debugCr(reconciliation, "Pod {} in namespace {} already exists => nothing to do right now", pod.getMetadata().getName(), reconciliation.namespace());
+
+            if (!ModelUtils.hasOwnerReference(currentPod, owner))  {
+                LOGGER.debugCr(reconciliation, "Pod {} in namespace {} is missing owner reference => patching it", currentPod.getMetadata().getName(), reconciliation.namespace());
+                Pod podWithOwnerReference = new PodBuilder(currentPod).build();
+
+                if (podWithOwnerReference.getMetadata().getOwnerReferences() != null)   {
+                    //List<OwnerReference> owners = new ArrayList<>()
+                    podWithOwnerReference.getMetadata().getOwnerReferences().add(owner);
+                } else {
+                    podWithOwnerReference.getMetadata().setOwnerReferences(List.of(owner));
+                }
+
+                podOperator.client().inNamespace(reconciliation.namespace()).patch(podWithOwnerReference);
+            }
```

Comment:
Is it possible for two Kafka CRs to map to the same `KafkaNodeSet` (e.g. if the selector labels of one are a subset of the other)?

Reply:
The selector is currently among other things  based on the name of the KafkaNodeSet. So no, it is not possible. Plus the same selector is used today with StatefulSets or Deployments where it would cause issues as well if there were any conflicts.

Reply:
@tombentley do you see any specific use case for doing that?

Reply:
No, I was just asking in case it was something we needed specifically to handle. But it doesn't sound like it.